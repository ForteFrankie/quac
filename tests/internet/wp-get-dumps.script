#!/bin/bash

# Test that wp-get-dumps script works. To do so, we download dumps for one of
# the smallest Wikipedias, Afar (http://aa.wikipedia.org).
#
# While the wiki is "closed" as of 4/23/2013, the content of the dump does
# change from time to time as the export format is updated.
#
# Copyright (c) Los Alamos National Security, LLC, and others.

. ./environment.sh

# Use a generic location rather than test-specific so that we can configure
# the path.
cd $TMPDIR
rm -Rf wp-dumps
mkdir wp-dumps

# Do the mirror. The head pipe is to remove the statistics, which change.
y "wp-get-dumps 2>&1 | head -n -15"

# Dumps contain timestamps and system version information which change.
# Therefore, we need to extract the .gz files and do a little magic for a
# proper comparison.
x cd wp-dumps/aawiki/latest
y "zcat aawiki-latest-pagelinks.sql.gz | egrep -v '^--' > aawiki-latest-pagelinks.sql"
y "zcat aawiki-latest-stub-meta-history.xml.gz | fgrep -v '<generator>' > aawiki-latest-stub-meta-history.xml"
x head -5 aawiki-latest-pagelinks.sql
x tail -5 aawiki-latest-pagelinks.sql
x head -5 aawiki-latest-stub-meta-history.xml
x tail -5 aawiki-latest-stub-meta-history.xml
