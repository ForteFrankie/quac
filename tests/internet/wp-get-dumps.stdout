$ (wp-get-dumps --notimes --config quacrc 2>&1 | head -n -15 | fgrep -v 'updating Wikimedia dumps')
wpdmp INFO     bandwidth limit is 10000 KB/s
receiving incremental file list
./
aawiki/
aawiki/latest/
aawiki/latest/aawiki-latest-pagelinks.sql.gz
aawiki/latest/aawiki-latest-stub-meta-history.xml.gz

Number of files: 5 (reg: 2, dir: 3)
Number of created files: 4 (reg: 2, dir: 2)
$ cd wp-dumps/aawiki/latest
$ (zcat aawiki-latest-pagelinks.sql.gz | egrep -v '^--' > aawiki-latest-pagelinks.sql)
$ (zcat aawiki-latest-stub-meta-history.xml.gz | fgrep -v '<generator>' > aawiki-latest-stub-meta-history.xml)
$ head -5 aawiki-latest-pagelinks.sql

/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!40101 SET NAMES utf8 */;
$ head -5 aawiki-latest-stub-meta-history.xml
<mediawiki xmlns="http://www.mediawiki.org/xml/export-0.9/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mediawiki.org/xml/export-0.9/ http://www.mediawiki.org/xml/export-0.9.xsd" version="0.9" xml:lang="aa">
  <siteinfo>
    <sitename>Wikipedia</sitename>
    <dbname>aawiki</dbname>
    <base>http://aa.wikipedia.org/wiki/Main_Page</base>
$ md5sum aawiki-latest-pagelinks.sql aawiki-latest-stub-meta-history.xml
c09541c752627b1485ade2cd3b2bdd13  aawiki-latest-pagelinks.sql
c0f559df40e6bcfaff14761e7afb4067  aawiki-latest-stub-meta-history.xml
